syntax = "proto3";

import "google/protobuf/empty.proto";
import "p2sro/p2sro_manager/protobuf/p2sro_manager.proto";

message PolicySpecJsonList {
    repeated float policy_prob_list = 1;
    repeated PolicySpecJson policy_spec_json_list = 2;
}

service PSRODistillManager {
    rpc CheckNumPlayers (NumPlayers) returns (Confirmation) {}
    rpc GetManagerMetaData (google.protobuf.Empty) returns (Metadata) {}
    rpc GetLogDir (google.protobuf.Empty) returns (String) {}
    rpc ClaimNewActivePolicyForPlayer (NewActivePolicyRequest) returns (PolicySpecJson) {}
    rpc SubmitNewActivePolicyMetadata (PolicyMetadataRequest) returns (PolicySpecJson) {}
    rpc CanActivePolicyBeSetAsFixedNow (PlayerAndPolicyNum) returns (Confirmation) {}
    rpc IsPolicyFixed (PlayerAndPolicyNum) returns (Confirmation) {}
    rpc SetActivePolicyAsFixed (PolicyMetadataRequest) returns (PolicySpecJson) {}
    rpc GetCopyOfLatestData (google.protobuf.Empty) returns (P2SROStatusResponse) {}
    rpc SubmitEmpiricalPayoffResult (PayoffResult) returns (Confirmation) {}
    rpc RequestExternalEval (EvalRequest) returns (Confirmation) {}
    rpc GetDistilledMetaNash (PolicySpecJsonList) returns (PolicySpecJson) {}
}